src/tables.rs

```diff
@@ -387,9 +387,81 @@ pub mod grapheme {
     }
 
     pub fn grapheme_category(c: char) -> (u32, u32, GraphemeCat) {
-        bsearch_range_value_table(c, grapheme_cat_table)
+        let idx = c as usize / 0x80;
```

Comment:
please add/generate comments into the code that explains what's going on here

also on the const

Reply:
Sounds good! Force pushed with comments!